crates/core/src/factories/core_space/test.rs

```diff
@@ -214,3 +214,94 @@ async fn space_notify_send_recv() {
     assert_eq!(TEST_SPACE, s);
     assert_eq!("world", String::from_utf8_lossy(&d));
 }
+
+// this is a bit of an integration test...
+// but the space module is a bit of an integration module...
+#[tokio::test(flavor = "multi_thread")]
+async fn space_local_agent_periodic_re_sign_and_bootstrap() {
+    #[derive(Debug)]
+    struct B(pub Mutex<Vec<Arc<agent::AgentInfoSigned>>>);
+
+    impl bootstrap::Bootstrap for B {
+        fn put(&self, info: Arc<agent::AgentInfoSigned>) {
+            self.0.lock().unwrap().push(info);
+        }
+    }
+
+    #[derive(Debug)]
+    struct BF(pub Arc<B>);
+
+    impl bootstrap::BootstrapFactory for BF {
+        fn default_config(&self, _config: &mut config::Config) -> K2Result<()> {
+            Ok(())
+        }
+
+        fn create(
+            &self,
+            _builder: Arc<builder::Builder>,
+            _peer_store: peer_store::DynPeerStore,
+            _space: SpaceId,
+        ) -> BoxFut<'static, K2Result<bootstrap::DynBootstrap>> {
+            let out: bootstrap::DynBootstrap = self.0.clone();
+            Box::pin(async move { Ok(out) })
+        }
+    }
+
+    #[derive(Debug)]
+    struct S;
+
+    impl SpaceHandler for S {}
+
+    #[derive(Debug)]
+    struct K;
+
+    impl KitsuneHandler for K {
+        fn create_space(
+            &self,
+            _space: SpaceId,
+        ) -> BoxFut<'_, K2Result<space::DynSpaceHandler>> {
+            Box::pin(async move {
+                let s: DynSpaceHandler = Arc::new(S);
+                Ok(s)
+            })
+        }
+    }
+
+    let b = Arc::new(B(Mutex::new(Vec::new())));
+
+    let builder = builder::Builder {
+        verifier: Arc::new(TestVerifier),
+        bootstrap: Arc::new(BF(b.clone())),
+        ..crate::default_builder()
+    }
+    .with_default_config()
+    .unwrap();
+
+    builder
+        .config
+        .set_module_config(&super::CoreSpaceModConfig {
+            core_space: super::CoreSpaceConfig {
+                // check every 5 millis if we need to re-sign
+                re_sign_freq_ms: 5,
+                // set it so we have to re-sign ALL agent infos always
```

Comment:
This is setting the expire value to be equal to the period of time the agent info is valid for so that work always needs doing? Could the comment describe that, that the value is chosen relative to the agent info validity period?

Reply:
Let me know if this is better: https://github.com/holochain/kitsune2/pull/61/commits/93f7e03e60d661c77de075e94b763564312541ea

Reply:
Awesome, yes thank you!